src/auth/toUtf.cc

```diff
@@ -58,7 +58,7 @@ Cp1251ToUtf8(const char *in)
 
         if (ch < 0x80)
             u = ch;
-        else if (ch >= 0xC0 && ch <= 0xFF) // 0x0410..0x044F
```

Comment:
> char is not guaranteed to be uint8_t in Windows and maybe elsewhere.

FYI: sizeof `char` is [always](https://isocpp.org/wiki/faq/intrinsic-types#sizeof-char) one [byte](https://isocpp.org/wiki/faq/intrinsic-types#sizeof-is-bytes). I bet Squid has several places that rely on that invariant.

Reply:
The error was that the second part of the condition was always true.

It would be interesting to see what happens on platforms where a byte is not 8 bits, but is there any such platform anymore?

Reply:
> The _bit_ size of char is the issue. [1 byte != 8 bits](https://isocpp.org/wiki/faq/intrinsic-types#bits-per-byte).

Good point!
 
> What I interpret as the real fix is to try detecting the systems 8-bit / 32-bit / 64-bit type and restructure the C++ if-else statement with #if statements. as needed for the bits in ch.

I bet Squid does not support systems with bytes exceeding 8 bits in size, for many reasons. Instead of adding complex and nearly impossible to test #ifdef code, we should simply statically assert that char has 8 bits (and still remove the always-true 0xFF comparison on the grounds of that assertion).

```C++
static_assert(std::numeric_limits<unsigned char>::max() == 0xFFu, "8-bit chars require no runtime 0xFF tests");
```

AFAICT, the above assertion uses a constant expression so, I hope that GCC will not complain about it:

```
-Wtype-limits
    Warn if a comparison is always true or always false due to the limited range of the data type, but do not warn for constant expressions.
```

Reply:
We do want to support Windows native builds. But that static_assert will be sufifcient for this PR.

Reply:
Windows is a red herring in this change request:

1. I still believe that Squid (in general, not just this piece of official and PR code) does not support environments where `char` is not exactly 8 bits in size. Adding the corresponding explicit static assertion will not change the set of supported environments but will allow us to safely remove the condition that is always true in any supported environment.
2. Native Windows compilers use 8-bit chars so the original fear itself is misplaced. You were probably thinking about internal text character representation, wide characters, and/or other things unrelated to the C/C++ type `char`. More from the horse's mouth at https://docs.microsoft.com/en-us/cpp/cpp/char-wchar-t-char16-t-char32-t

Reply:
@kinkie, your [statement](https://github.com/squid-cache/squid/pull/667#issuecomment-890356040) seems to match my understanding and my specific [suggestion](https://github.com/squid-cache/squid/pull/667/files#r679930222) on how to address this change request. Amos liked that suggestion. Does anything prevent you from implementing it?

Reply:
@kinkie, I fixed and polished the corresponding change. Please see 7d0c17a and adjust further if needed.